### PR TITLE
Add bwip-js w/ npm auto-update

### PR DESCRIPTION
**Pull request by hare1039**
Pull request for issue: #10215
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/metafloor/bwip-js
 * Official website (optional, not the repository): http://bwip-js.metafloor.com/
 * NPM package url (optional): https://www.npmjs.com/package/bwip-js
 * License and its reference: MIT, [ref](https://github.com/metafloor/bwip-js/blob/master/LICENSE)
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 34
   - Count of stars: 607
   - Count of forks: 73
 * NPM download stats (optional):
   - Downloads in the last day: 451
   - Downloads in the last week: 2127
   - Downloads in the last month: 9392

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by hare1039** [comment]
@cdnjs/intern3 Please help me review this PR. Thank you.
Note I only include `bwipjs.js` because `bwipp` is another [project](https://github.com/bwipp/postscriptbarcode)
http://bwip-js.metafloor.com/#user-content-installation

**Comment by hare1039** [comment]
@sufuf3 I added version 0.6-0.9. Please review this PR again. Thanks.

**Comment by hare1039** [comment]
@pvnr0082t I've rebased to master branch. Please help me review this PR. Thank you.

File: ajax/libs/bwip-js/0.6/bwip.js [hunk 1]
```diff
@@ -0,0 +1,991 @@
...
+		size = 10;
+	else if (size > 10)
+		size = 12;
+
+	var key = size + (fs < 10 ? '-0' : '-') + fs;
+
+	// Has the font been loaded?
+	if (!BWIPJS.fonts.OCRB || !BWIPJS.fonts.OCRB[key])
+		// Invoke the file loader
+		BWIPJS.load('fonts/ocrb' + key + '.js');
```
Comment (pvnr0082t) [inline ajax/libs/bwip-js/0.6/bwip.js]: Don't we need files under `fonts`?

File: ajax/libs/bwip-js/package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  ],
+  "homepage": "http://bwip-js.metafloor.com/",
+  "author": "Mark Warren <mwarren@metafloor.com>",
+  "license": "MIT",
+  "npmName": "bwip-js",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "bwipjs.js"
```
Comment (pvnr0082t) [inline ajax/libs/bwip-js/package.json]: `"bwipjs*"` will be better.

**Comment by hare1039** [comment]
@pvnr0082t I add 
`fonts/*.js` <= https://github.com/metafloor/bwip-js/blob/v0.6/bwip.js#L491
`bwipp/*.js` <= https://github.com/metafloor/bwip-js/blob/v0.6/bwip.js#L303
and amend the commit massage.
Please help me review this PR again. Thanks.

**Comment by hare1039** [comment]
@PeterBot Thanks!

@pvnr0082t Please review this PR again. Thanks.

**Comment by hare1039** [comment]
@pvnr0082t I fix the names of old versions. Please review this PR again. Thanks.

**Comment by PeterDaveHello** [comment]
@hare1039 you should select the reviewer again instead of just leaving comment.

**Comment by hare1039** [comment]
@pvnr0082t I fixed the commit message. Please review this PR again. Thanks.

**Comment by hare1039** [comment]
@extend1994 GitHub's version list is totally different from npm. So both method need to add version manually.

**Comment by hare1039** [comment]
@pvnr0082t I rebased it to latest master mranch.

**Comment by PeterDaveHello** [comment]
Ping @sufuf3

**Comment by PeterDaveHello** [comment]
Why don't you make the name same as what it is? What is the reason why to use`bwipjs` in PR title and commit message, but `bwip-js` in the `package.json`?

**Comment by hare1039** [comment]
@pvnr0082t I change the commit message as request. Please review this PR again. Thanks.

**Comment by PeterDaveHello** [comment]
@hare1039 you mean you "fix the inconsistency",

**Comment by PeterDaveHello** [comment]
https://github.com/cdnjs/cdnjs/pull/10581#issuecomment-278908627

> @cdnjs/intern3 Please help me review this PR. Thank you.
> Note I only include bwipjs.js because bwipp is another project
> ref: http://bwip-js.metafloor.com/#user-content-installation

but why I see `bwipp-min.js` and `bwipp.js` in the commit?  I didn't see you update any related news here, even though you opened an issue to upstream, you should report the update here.

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by hare1039** [comment]
@pvnr0082t Should I use (`filename: bwipjs.js` which `bwipjs.js` is not minified) in this case?

**Comment by pvnr0082t** [comment]
@hare1039 Yes, I think we can do this.

**Comment by hare1039** [comment]
@pvnr0082t I removed minified files and rebased with master. Please review this PR again. Thanks.

**Comment by PeterDaveHello** [comment]
Why do you use a different homepage url in comment and package.json?

**Comment by PeterDaveHello** [comment]
It's a complex case here ... partially minified with different naming scheme, so have you asked about the minified files from upstream?

**Comment by PeterDaveHello** [comment]
Ping @hare1039

**Comment by sufuf3** [comment]
@hare1039 Please try to answer the comment https://github.com/cdnjs/cdnjs/pull/10581#issuecomment-297485986
Also, we like to know the answer.

Thank you.

**Comment by sufuf3** [comment]
@hare1039 Would you like to work on this and help improve this PR after the final exam?

Many thanks.

**Comment by sufuf3** [comment]
Help ping @hare1039

**Comment by sashberd** [comment]
Moved to #13711

**Pull request closed without merging**